scrapbook/api.py

```diff
@@ -19,6 +17,24 @@
 from .utils import kernel_required
 
 
+def determine_encoder(scrap):
+    # Keep slow import lazy
+    import IPython
+    from IPython.display import DisplayObject
+
+    if isinstance(scrap, str):
+        return "text"
+    elif isinstance(scrap, (list, dict)):
+        return "json"
+    elif isinstance(scrap, DisplayObject):
+        return "display"
+    elif isinstance(scrap, pd.DataFrame):
+        return "pandas"
+    else:
```

Comment:
Should we add a null-case here to avoid `scrap=None` being encoded as JSON and triggering undefined cases?

Reply:
Yes. I'll just go ahead and change the behavior to raise in this case. Json conversion would be likely to fail anyway.